src/pages/quotes/QuotePage.tsx

```diff
@@ -1,24 +1,35 @@
 import { Breadcrumbs } from "@/components/Breadcrumbs"
 import { PageTitle } from "@/components/PageTitle"
-import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/avatar"
+import { Avatar } from "@/components/ui/avatar"
 import { Badge } from "@/components/ui/badge"
 import { Button } from "@/components/ui/button"
 import { Skeleton } from "@/components/ui/skeleton"
 import { Table, TableBody, TableCell, TableRow } from "@/components/ui/table"
 import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from "@/components/ui/tooltip"
-import { IdentityPublicData, InfoReply } from "@/generated/client"
+import { IdentityPublicData, PayeePublicData, InfoReply, AnonPublicData } from "@/generated/client"
 import {
   adminLookupQuoteOptions,
   adminLookupQuoteQueryKey,
   adminUpdateQuoteMutation,
 } from "@/generated/client/@tanstack/react-query.gen"
 import { activateKeyset } from "@/generated/client/sdk.gen"
-import useLocalStorage from "@/hooks/use-local-storage"
 import { cn } from "@/lib/utils"
 import { formatDate, humanReadableDuration } from "@/utils/dates"
-import { randomAvatar } from "@/utils/dev"
+
 import { formatNumber, truncateString } from "@/utils/strings"
 import { useMutation, useQueryClient, useSuspenseQuery } from "@tanstack/react-query"
+import { getDeterministicColor } from "@/utils/dev"
+
+// Utility functions for deterministic avatars
+function getInitials(name?: string): string {
+  if (!name) return "?"
+  const words = name.trim().split(/\s+/)
+  if (words.length === 1) {
+    return words[0].substring(0, 2).toUpperCase()
+  }
+  return (words[0][0] + words[words.length - 1][0]).toUpperCase()
+}
```

Comment:
Maybe move this to it's own utils file? Not sure if it's an early optimization tho, please tell me. Also, we have this very same utility with a cleaner functional approach in our frontend repo:

https://github.com/BitcreditProtocol/E-Bill-frontend/blob/main/src/utils/strings.ts#L11

We have other util functions there too, so please feel free to use as many of those as you want to!